src/Query/Query.php

```diff
@@ -385,6 +387,19 @@ public function filterWhere(array $condition): static
         return $this;
     }
 
+    public function for(ExpressionInterface|string|null $value = null): static
```

Comment:
```suggestion
    public function for(string|null $value = null): static
```

I think no need `ExpressionInterface` here. It should be `string` or `null` and possible constants should be provided,
for example `SelectFor::UPDATE`

Reply:
`FOR` clause may be difficult.

For example, [PostgreSQL](https://postgrespro.ru/docs/postgresql/17/sql-select):
```
SELECT ... 
[ 
	FOR { UPDATE | NO KEY UPDATE | SHARE | KEY SHARE } 
	[ OF _`from_reference`_ [, ...] ]
	[ NOWAIT | SKIP LOCKED ] 
	[...]
]
```

… or [MySQL](https://dev.mysql.com/doc/refman/8.4/en/select.html):

```
SELECT ... 
[
	FOR {UPDATE | SHARE}
	[OF _tbl_name_ [, _tbl_name_] ...] 
	[NOWAIT | SKIP LOCKED] | LOCK IN SHARE MODE
]
```

So, I think expression is expedient here.

Reply:
The `string` is the `expression`. I do not see reasons to use `ExpressionInterface`.

It looks like index type or index method from `CommandInterface::createIndex()`

https://github.com/yiisoft/db/blob/08fe349c30ba710da793fc359bb89cc1fb87db88/src/Command/CommandInterface.php#L310-L316

Reply:
In future we can create special expression objects, that simplify building difficult DBMS-specific `FOR` clause with automatic table/column names quoting. WDYT?

Reply:
> In future we can create special expression objects, that simplify building difficult DBMS-specific `FOR` clause with automatic table/column names quoting. WDYT?

Not sure, do you have an example?
 
> ```sql
> SELECT * FROM t1, t2 FOR SHARE OF t1 FOR UPDATE OF t2;
> ```

This can be done by
`$query->for('SHARE OF t1 FOR UPDATE OF t2');`

Reply:
OK, I removed expression support.

But array of values is keeps.  "FOR UPDATE ..." is another "for" part

```php
$query->for('SHARE OF {{t1}} FOR UPDATE OF {{t2}}');

// seems, it's more logically
$query->for([
  'SHARE OF {{t1}}',
  'UPDATE OF {{t2}}',
]);
```